Components/CategoryResolver.php

```diff
@@ -129,10 +129,7 @@ private function removeProductsFromRemoteCategory(array $assignedCategories, $ar
         /** @var int $currentProductCategoryId */
         foreach ($currentProductCategoryIds as $currentProductCategoryId) {
             if (!in_array($currentProductCategoryId, $assignedCategoryIds)) {
-                $this->productToRemoteCategoryRepository->deleteByConnectCategoryId($currentProductCategoryId);
-                if ($this->productToRemoteCategoryRepository->getProductCountByCategoryId($currentProductCategoryId) > 0) {
-                    $this->remoteCategoryRepository->deleteById($currentProductCategoryId);
-                }
+                $this->productToRemoteCategoryRepository->deleteByConnectCategoryId($currentProductCategoryId, $articleId);
```

Comment:
Why you removed the check where cont if there is at least 1 product in remote category? Otherwise remove it completely.

Reply:
FUCK!!!!!!!!
Because it made trouble 
the greater than is completly bullshit it has to be less or equals

Reply:
still makes trouble with the correct condition
I created a follow-up:
https://jira.shopware.com/browse/CON-4885